### PR TITLE
Enable UseShellExecute to fix LaunchUriOrFileAction

### LINKED ISSUE #8: LaunchUriOrFileAction fails in .NET Core 3
Using  `LaunchUriOrFileAction` with WPF on NETCore3 Preview 1 throws the exception below:

```
System.ComponentModel.Win32Exception
  HResult=0x80004005
  Message=The system cannot find the file specified.
  Source=System.Diagnostics.Process
  StackTrace:
   at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start(String fileName)
   at Microsoft.Xaml.Behaviors.Core.LaunchUriOrFileAction.Invoke(Object parameter)
   at Microsoft.Xaml.Behaviors.TriggerAction.CallInvoke(Object parameter)
   at Microsoft.Xaml.Behaviors.TriggerBase.InvokeActions(Object parameter)
   at Microsoft.Xaml.Behaviors.EventTriggerBase.OnEvent(EventArgs eventArgs)
   at Microsoft.Xaml.Behaviors.EventTriggerBase.OnEventImpl(Object sender, EventArgs eventArgs)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.UIElement.RaiseEvent(RoutedEventArgs e)
   at System.Windows.Controls.Primitives.ButtonBase.OnClick()
   at System.Windows.Controls.Button.OnClick()
   at System.Windows.Controls.Primitives.ButtonBase.OnMouseLeftButtonUp(MouseButtonEventArgs e)
   at System.Windows.UIElement.OnMouseLeftButtonUpThunk(Object sender, MouseButtonEventArgs e)
   at System.Windows.Input.MouseButtonEventArgs.InvokeEventHandler(Delegate genericHandler, Object genericTarget)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.EventRoute.ReInvokeHandlers(Object source, RoutedEventArgs args)
   at System.Windows.UIElement.ReRaiseEventAs(DependencyObject sender, RoutedEventArgs args, RoutedEvent newEvent)
   at System.Windows.UIElement.CrackMouseButtonEventAndReRaiseEvent(DependencyObject sender, MouseButtonEventArgs e)
   at System.Windows.UIElement.OnMouseUpThunk(Object sender, MouseButtonEventArgs e)
   at System.Windows.Input.MouseButtonEventArgs.InvokeEventHandler(Delegate genericHandler, Object genericTarget)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
   at System.Windows.Input.InputManager.ProcessStagingArea()
   at System.Windows.Input.InputManager.ProcessInput(InputEventArgs input)
   at System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
   at System.Windows.Interop.HwndMouseInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawMouseActions actions, Int32 x, Int32 y, Int32 wheel)
   at System.Windows.Interop.HwndMouseInputProvider.FilterMessage(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.InputFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.TranslateAndDispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.Run()
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at System.Windows.Application.Run(Window window)
   at System.Windows.Application.Run()
   at BehaviorsOnCore.App.Main()

```

### PR DESCRIPTION
**Pull request by Lakritzator**
Although dotnet core 3.0 is not supported, this change will at least solve #8 
I can image that there are cases where UseShellExecute doesn't make sense, with UWP this should be false.
But this library is only for .NET Framework and has a potential target in dotnet core 3.0, so from my point of view the fix doesn't to be more complex... just enable it!

### CODE REVIEW

**Comment by brianlagunas** [comment]
The default for UseShellExecute is true on .NET Framework apps and false on .NET Core apps. So technically, this will have no impact on existing .NET Framework projects using it.

I would also add that setting the `CreateNoWindow` property has no value here because if `UseShellExecute` is true, the `CreateNoWindow` property is ignored and a new window is created.  I'm not sure if that holds true for .NEt Core 3 apps. I would have to test this.

However, given that we do not officially support .NET Core 3, I am wondering if this should even be added. Maybe it would be best to wait until we have a target for .NET Core 3 before trying to "fix" #8, as that is really due to a lack of .NET Core 3 support.

**Comment by dotMorten** [comment]
How about I add a netcore3.0 Target and to start exclude this class? (And anything else that's not supported and not an easy fix)

**Comment by brianlagunas** [comment]
That's kind of where I was going with my comment. Just add support for .NET Core 3 and then handle this.

**Comment by Lakritzator** [comment]
I'm fine with whatever you decide, just found a quick sensible solution for the reported issue.

**Comment by PedroLamas** [comment]
Instead of changing the behavior, why not just add properties for `UseShellExecute` and `CreateNoWindow` so consumers can change as they wish?

**Comment by dotMorten** [comment]
@pedrolamas that still doesn't change the fact that the default is different on the two platforms. Makes sense to me to force it (or the very least set it in an if-def for netcoreapp3.0, but there really no point since it would just be a no op on netfx)

**Comment by dotMorten** [comment]
I think we need a unit test to test this - All tests pass on .NET Core 3.0, so this must not be covered.

File: src/Microsoft.Xaml.Behaviors/Core/LaunchUriOrFileAction.cs [hunk 1]
```diff
@@ -34,8 +34,13 @@ public string Path
         protected override void Invoke(object parameter)
         {
             if (this.AssociatedObject != null && !string.IsNullOrEmpty(this.Path))
-            {
-                Process.Start(this.Path);
+            {
+                var processStartInfo = new ProcessStartInfo(this.Path)
+                {
+                    CreateNoWindow = true,
```
Comment (dotMorten) [inline src/Microsoft.Xaml.Behaviors/Core/LaunchUriOrFileAction.cs]: Why this change? The default is false on both WPF and .NET Core. I think it's safer to not set this, so this doesn't change behavior, and it seems to continue to work without setting it.

File: src/Microsoft.Xaml.Behaviors/Core/LaunchUriOrFileAction.cs [hunk 2]
```diff
@@ -34,8 +34,13 @@ public string Path
         protected override void Invoke(object parameter)
         {
             if (this.AssociatedObject != null && !string.IsNullOrEmpty(this.Path))
-            {
-                Process.Start(this.Path);
+            {
+                var processStartInfo = new ProcessStartInfo(this.Path)
+                {
+                    CreateNoWindow = true,
+                    UseShellExecute = true
```
Comment (dotMorten) [inline src/Microsoft.Xaml.Behaviors/Core/LaunchUriOrFileAction.cs]: This looks good to me. Confirmed this is the default on NETFX already, and is false on `netcoreapp3.0`, so it's essentially a no-op and enables some netcore compatibility.

**Comment by PedroLamas** [comment]
> that still doesn't change the fact that the default is different on the two platforms

Indeed, you are correct. Yet, I still maintain my point: why not just add properties for `UseShellExecute` and `CreateNoWindow` so consumers can change as they wish?

**Comment by brianlagunas** [comment]
ProcessInfo has 22 properties. Where do we draw the line of surfacing properties?  I would be hesitant to open that Pandora's box.

**Comment by dotMorten** [comment]
>  Yet, I still maintain my point: why not just add properties for UseShellExecute and CreateNoWindow so consumers can change as they wish

Do you have a use-case / user-story to support exposing those properties? Why would you want/need to set them? Aren't the defaults sufficient? (provided we make the default correct for .NET Core)

I personally don't like adding members "just in-case". More properties means more to doc, more to understand, more to be confused by, and more ways to shoot yourself in the foot. IMHO each member should have a well-understood and reasonably common scenario to support them.

**Comment by dotMorten** [comment]
> However, given that we do not officially support .NET Core 3, I am wondering if this should even be added. 

I think it should. Microsoft has been bragging about how your generally can use .NET Framework assemblies "as-is" in .NET Core 3.0 without a recompilation needed. This is a small safe change to make that work, without worrying about compiling for .NET Core 3.0 explicitly.

File: src/Microsoft.Xaml.Behaviors/Core/LaunchUriOrFileAction.cs (hunk 1 above)
Comment (Lakritzator) [reply src/Microsoft.Xaml.Behaviors/Core/LaunchUriOrFileAction.cs]: I guess I overlooked the default on .NET Core, it's not needed. I removed it!

**Comment by brianlagunas** [comment]
Thanks for the PR
